sdk/core/azure-core/tests/test_authentication.py

```diff
@@ -146,3 +146,39 @@ def test_key_vault_regression():
     policy._token = AccessToken(token, time.time() + 3600)
     assert not policy._need_new_token
     assert policy._token.token == token
+
+def test_azure_key_credential_policy():
+    """Tests to see if we can create an AzureKeyCredentialPolicy"""
+    def verify_authorization_header(request):
+        assert request.headers["api_key"] == "test_key"
+    api_key = "test_key"
+    key_header = "api_key"
```

Comment:
You can leverage Python's ~~sloppy~~ flexible scoping to better use these variables:
```suggestion
    api_key = "test_key"
    key_header = "api_key"

    def verify_authorization_header(request):
        assert request.headers[key_header] == api_key
```